### PR TITLE
chore: bump kubectl version in prow-test image

### PR DESCRIPTION
**Pull request by Cali0707**
**What this PR does, why we need it**:<br>
<Explanation goes here>
This PR bumps the kubectl version we use in the prow test images to a newer version so that it is more aligned with the kubernetes version we use in the tests

### CODE REVIEW

File: images/prow-tests/Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ FROM debian:bullseye-20240211-slim AS base
 # Pinned versions of stuff we pull in
 ARG TARGETARCH
 ARG CLOUD_SDK_VERSION=432.0.0
-ARG KUBECTL_VERSION=v1.25.12
+ARG KUBECTL_VERSION=v1.28.7
```
Comment (creydr) [inline images/prow-tests/Dockerfile:20]: we could also go directly to 1.29.x, as the client is backwards compatible for some versions AFAIK

Comment (upodroid) [reply images/prow-tests/Dockerfile:20]: happy with either option

Comment (Cali0707) [reply images/prow-tests/Dockerfile]: I updated it to 1.29.2, nice idea @creydr !

**Review by creydr — APPROVED** [review]
/lgtm

**Pull request merged**